pkg/sysinfo/cgroup2_linux.go

```diff
@@ -66,15 +67,22 @@ func newV2(quiet bool, opts *opts) *SysInfo {
 	return sysInfo
 }
 
-func applyMemoryCgroupInfoV2(info *SysInfo, controllers map[string]struct{}, _ string) []string {
+func applyMemoryCgroupInfoV2(info *SysInfo, controllers map[string]struct{}, dirPath string) []string {
 	var warnings []string
 	if _, ok := controllers["memory"]; !ok {
 		warnings = append(warnings, "Unable to find memory controller")
 		return warnings
 	}
 
+	info.SwapLimit = false
+	if "/sys/fs/cgroup" != path.Clean(dirPath) {
```

Comment:
Actually, if `/sys/fs/cgroup/memory.swap.max` never appears, this check right now could be that `path.Join(dirPath, "memory.swap.max")` exists, without checking that we are in the group. Let me know what you think

Reply:
This does not work when dirPath == /sys/fs/cgroup

Reply:
Right now if /sys/fs/cgroup then it is assumed upper function did not deliver /sys/fs/cgroup/${dockerd_cgroup}/ and info.SwapLimit = false defaults to false. Is that wrong?

Reply:
The caller never pass "/sys/fs/cgroup/${dockerd_cgroup}". (See `git grep WithCgroup2GroupPath`).

So we need to detect the cgroup of the daemon by ourselves, regardless to the value of `dirPath`.

Reply:
> The caller never pass "/sys/fs/cgroup/${dockerd_cgroup}". (See git grep WithCgroup2GroupPath).

Sorry, I thought `g` in the caller is the added part that will complement the path with the group. 
Sysinfo can be used in a package that does not belong to dockerd, so /proc/self/cgroup is not the option here, am I right?

Reply:
I think we can safely use /proc/self/cgroup here